### DISCUSSION TITLE
Change response_model parameter to type hint of return value

### DISCUSSION THREAD

**Discussion by Vitalium** [fastapi/fastapi#5485] [Questions] [answered]
Hello!

My proposition is to change routes declaration to be more pythonic. For example, instead of:
```python
@app.post("/user/", response_model=UserOut)
async def create_user(user: UserIn):
    return user
```
use this:
```python
@app.post("/user/")
async def create_user(user: UserIn) -> UserOut:
    return user
```

I think it looks nice, intuitive and obvious :)

Inside `api_route` decorator we can get response_model model like this:
`response_model  = func.__annotation__.get('return', response_model)`

I understand that it only covers simple cases, but I think it's worth implementing, since it's done with just one line of code and supports backward compatibility.

**Comment by nymous** [comment]
This is regularly suggested, see #101, #2296, #5215 (and probably others), but it's hard to do. The rationale is documented in the "Technical details" box above this title https://fastapi.tiangolo.com/tutorial/response-model/#return-the-same-input-data
> The response model is declared in this parameter instead of as a function return type annotation, because the path function may not actually return that response model but rather return a `dict`, database object or some other model, and then use the `response_model` to perform the field limiting and serialization.

If you want this feature, `fastapi-utils` has an `InferringRouter`: https://fastapi-utils.davidmontague.xyz/user-guide/inferring-router/

**Comment by tiangolo** [comment, marked as answer]
This is now supported: https://fastapi.tiangolo.com/tutorial/response-model/ :tada: 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
